source/common/config/new_grpc_mux_impl.cc

```diff
@@ -153,13 +155,28 @@ GrpcMuxWatchPtr NewGrpcMuxImpl::addWatch(const std::string& type_url,
 // subscription will enqueue and attempt to send an appropriate discovery request.
 void NewGrpcMuxImpl::updateWatch(const std::string& type_url, Watch* watch,
                                  const std::set<std::string>& resources,
-                                 bool creating_namespace_watch) {
+                                 const bool creating_namespace_watch) {
   ASSERT(watch != nullptr);
   auto sub = subscriptions_.find(type_url);
   RELEASE_ASSERT(sub != subscriptions_.end(),
                  fmt::format("Watch of {} has no subscription to update.", type_url));
-  auto added_removed = sub->second->watch_map_.updateWatchInterest(watch, resources);
-  if (creating_namespace_watch) {
+  // If this is a glob collection subscription, we need to compute actual context parameters.
+  std::set<std::string> xdstp_resources;
+  if (resources.size() == 1 && XdsResourceIdentifier::hasXdsTpScheme(*resources.begin())) {
+    auto resource = XdsResourceIdentifier::decodeUrn(*resources.begin());
+    // We only know how to deal with glob collections right now.
```

Comment:
Should there be an error in an else clause even if you plan on implementing more later? It would be more clear on the future direction also.